[DOCS] Expands ELSER tutorial with optimization info (#97392)
elastic/elasticsearch: docs/reference/query-dsl/text-expansion-query.asciidoc

   }
}
----
// TEST[skip: TBD]

[discrete]
[[optimizing-text-expansion]]
=== Optimizing the search performance of the text_expansion query

> https://www.elastic.co/blog/faster-retrieval-of-top-hits-in-elasticsearch-with-block-max-wand[Max WAND] 

Reviewer: ```suggestion
https://www.elastic.co/blog/faster-retrieval-of-top-hits-in-elasticsearch-with-block-max-wand[Max WAND] is an
```

> optimization technique used by {es} to skip over documents that don't score 

Reviewer: ```suggestion
optimization technique used by {es} to skip documents that cannot score 
```

competitively against the current best matching documents. However, the weighted 

> terms generated by the ELSER model don't work well with the Max WAND 

Reviewer: The rest of the doc uses the phrase `token-weight pairs` rather than `weighted terms`, to be consistent `tokens` should be preferred to `terms`.

This could be rewritten as "However, the tokens generated by the ELSER...."

optimization. Consequently, enabling Max WAND can actually increase query 
latency for `text_expansion`. For datasets of a significant size, disabling Max 
WAND leads to lower query latencies.

> The behavior of Max WAND can be controlled using the 

Reviewer: ```suggestion
Max WAND is controlled by the
```

<<track-total-hits, track_total_hits>> query parameter. Setting track_total_hits 
to true forces {es} to consider all documents, resulting in lower query 
latencies for the `text_expansion` query. However, other {es} queries run slower 
when Max WAND is disabled.

If you are combining the `text_expansion` query with standard text queries in a 
compound search, it is recommended to measure the query performance before 
deciding which setting to use.

NOTE: The `track_total_hits` option applies to all queries in the search request 

> and may be optimal for some queries but not for others. Take into account of the 

Reviewer: ```suggestion
and may be optimal for some queries but not for others. Take into account the 
```
